src/mame/f256/tiny_vicky.cpp

```diff
@@ -0,0 +1,613 @@
+// license:BSD-3-Clause
+// copyright-holders:Daniel Tremblay
+#include "emu.h"
+#include "tiny_vicky.h"
+
+#include "machine/ram.h"
+#include "screen.h"
+
+DEFINE_DEVICE_TYPE(TINY_VICKY, tiny_vicky_video_device, "tiny_vicky", "F256K Tiny Vicky")
+
+tiny_vicky_video_device::tiny_vicky_video_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock):
+    device_t(mconfig, type, tag, owner, clock)
+    , m_sof_irq_handler(*this)
+    , m_sol_irq_handler(*this)
+{
+}
+
+tiny_vicky_video_device::tiny_vicky_video_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock):
+    tiny_vicky_video_device(mconfig, TINY_VICKY, tag, owner, clock)
+{
+}
+
+rgb_t tiny_vicky_video_device::get_text_lut(uint8_t color_index, bool fg, bool gamma)
+{
+    uint8_t red =   iopage0_ptr[(fg ? 0x1800 : 0x1840) + color_index * 4 + 2];
```

Comment:
Is `iopage0_ptr` really pointing to a palette or RAMDAC device here?

Reply:
It's a color Look Up Table (LUT), indeed like a palette.  But only 16 colors for foreground and 16 colors for background.  Each color is represented by 4 bytes: blue, green, red, and alpha (alpha is ignore by the machine so far).

Reply:
> It's a color Look Up Table (LUT), indeed like a palette. But only 16 colors for foreground and 16 colors for background. Each color is represented by 4 bytes: blue, green, red, and alpha (alpha is ignore by the machine so far).

Perfect use case for a `palette_device`. The final bitmap should be 32bpp, so you're solid in that regard, but iit's preferable to use a `palette_device` for even fairly baisc LUT remapping.

Reply:
The intent of the question was to prompt you to explain what the underlying hardware is, such that it could be emulated in a more realistic/accurate fashion. Functionally it's a LUT; what hardware is used to implement it?